File: spec/functional_test/fixtures/kotlin/cli_clikt/tool.kt
import com.github.ajalt.clikt.core.CliktCommand
import com.github.ajalt.clikt.core.subcommands
import com.github.ajalt.clikt.parameters.arguments.argument
import com.github.ajalt.clikt.parameters.options.option

class Tool : CliktCommand() {
    val verbose by option("-v", "--verbose")
    override fun run() {
        val token = System.getenv("API_TOKEN")
    }
}

class Serve : CliktCommand() {
    val port by option("-p", "--port")
    val config by argument()
    override fun run() {}
}

fun main(args: Array<String>) = Tool().subcommands(Serve()).main(args)

File: spec/functional_test/fixtures/clojure/cli_tools/tool.clj
(ns tool.core
  (:require [clojure.tools.cli :refer [parse-opts]]))

(def cli-options
  [["-p" "--port PORT" "Port number"]
   ["-v" "--verbose"]])

(defn -main [& args]
  (let [{:keys [options]} (parse-opts args cli-options)
        token (System/getenv "API_TOKEN")]
    [options token]))
